src/renderer/utils/hooks.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { FunctionReturnVoid, FunctionReturnPromiseVoid } from 'sharedTypes';
+
+function useKeypress(
+  eventHandler: FunctionReturnPromiseVoid | FunctionReturnVoid,
+  condition: boolean,
+  keypressCodes: string[]
+) {
+  useEffect(() => {
+    const handleKeypress = async (event: KeyboardEvent) => {
+      if (keypressCodes.includes(event.code) && !condition) {
```

Comment:
The hook looks much better. Nice!

One suggestion is change the input and usage of `condition`. It looks like the event handler only runs when `condition` is false, which seems a little confusing for other developers.

Reply:
Here is the commit with the change: [56ed2a8](https://github.com/chloebrett/mlvet/pull/175/commits/56ed2a81f3a3e2dce6254eed2d987f8738609f8f)